src/modules/complianceengine/src/lib/UsersIterator.h

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#ifndef COMPLIANCE_USERS_ITERATOR_H
+#define COMPLIANCE_USERS_ITERATOR_H
+
+#include <Logging.h>
+#include <MmiResults.h>
+#include <ReentrantIterator.h>
+#include <Result.h>
+#include <pwd.h>
+#include <vector>
+
+namespace ComplianceEngine
+{
+class UsersRange;
+class UsersIterator : public ReentrantIterator<struct passwd, UsersRange>
+{
+public:
+    explicit UsersIterator(const UsersRange* range)
+        : ReentrantIterator(range, fgetpwent_r)
```

Comment:
This just occurred to me: `fgetpwent_r` only parses the passed file, while `getpwent_r` also accesses NSS to enumerate users/groups.
Are these rules supposed to restrict themselves to only local userdbs? That depends on the groups that are passed to these probes.

Depending on the answer you may need to add 2 paths: one that parses a file (for mocking) and one that actually interacts with NSS.

Reply:
I'd stick to files too, with non-local users the policy should be checked elsewhere anyway - not on the machine.